src/backtrace.c

```diff
@@ -145,8 +147,10 @@ int backtrace_ptrace(int pid, int *tids, int *index, int nr_tids)
     for (i = 0; i < count; ++i) {
         void *upt_info;
 
-        printf("--------------------  thread %d (%d)  --------------------\n",
+        printf("--------------------  thread %d (%d) (",
                (index != NULL ? index[i] : i+1), threads[i]);
+        print_proc_comm(threads[i]);
```

Comment:
Is it a good idea to call print_proc_comm here, while the process is stopped?
As print_proc_comm is currently implemented it opens, reads, and closes the comm file for each thread.
In a large server process with thousands of threads, this will add noticeably to the time frozen.
Why not collect the thread names before the call to attach_process(), the same way thread states are handled, then pass the thread name for the current thread to print_thread_heading?